contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -3229,6 +3226,54 @@ class tsqlMutator : public TSqlParserBaseListener
 	{
 		handleModuloOperator(ctx);
 	}
+	
+	void enterUnary_op_expr(TSqlParser::Unary_op_exprContext *ctx) override
+	{
+		/* 
+		 * The T-SQL grammar allows an arbitrary number of unary '+' operators to precede an expression, 
+		 * but PG only supports that for numeric expressions. For string expressions, such a '+' will raise an error in PG.
+		 * In SQL this shows as redundant operators, for example for concatenation: SELECT 'a' ++ 'b'. Expressions
+		 * such as +++(+++@v)) are also valid syntax according to the T-SQL grammar even though they look unusual.
+		 * Here we remove such unary '+' operators, which are redundant anyway. 
+		 * However we do not touch numeric constants (e.g. +123) since the '+', although still redundant, may
+		 * have been included for code clarity (e.g. +123 as opposed to -123).
+		 */
+		std::string op = getFullText(ctx->op);	
+		if (op.front() == '+') {	
+			auto rhsctx = ctx->expression();
+			while (true) {						
+				std::string rhs = getFullText(rhsctx);
+				if (
+					(rhs.front() == '\'') ||    // single-quoted strings
+					(rhs.front() == '"')  ||    // both double-quoted strings and double-quoted identifiers
+					(rhs.front() == '@')  ||    // variables
+					(rhs.front() == '(')  ||    // bracketed expressions			
+					(rhs.front() == '[')  ||    // bracket-delimited identifiers
+					(rhs.front() == '_')  ||    // identifiers starting with an underscore
+					std::isalpha(rhs.front())   // identifiers as well as the N'...' string notation
```

Comment:
What are we trying to do here? Why only these characters are considered?

Reply:
These characters match anything except a numeric constant. Technically +(123) is also a numeric constant in T-SQL grammar terms, but at school we learned to interpret this as (+1)*(123) so removing the + is fine here, per the philosophy as outlined in the earlier comment above.

Reply:
So can't we check only if the next char is numeric? Something like this:
```
if (rhs.front() == '+')  {
   // do something
}
else if (!std::isdigit(rhs.front())) {
    stream.setText(ctx->op->getStartIndex(), " ");	
    break;
}
break;
```

Reply:
There would also have to be a test on the first char being '.' (e.g. +.5). 
The advantage of the current coding explains what sort of expression we're removing the '+' from, which I find more helpful. But that's partly a matter of taste.

Reply:
Yep, my point was to make the conditions a little simpler but anyways if it works this way then I am fine with it as well.